On macos 10.15:

```
builder for '/nix/store/x79m2v5jjijvlrpy9721pxfw74had61m-pdnsd-1.2.9a-par.drv' failed with exit code 2; last 10 log lines:
  #include <linux/if.h>
           ^~~~~~~~~~~~
  1 error generated.
  make[3]: *** [Makefile:339: pdnsd-conf-parser.o] Error 1
  make[3]: Leaving directory '/private/tmp/nix-build-pdnsd-1.2.9a-par.drv-0/pdnsd-1.2.9a/src'
  make[2]: *** [Makefile:607: all-recursive] Error 1
  make[2]: Leaving directory '/private/tmp/nix-build-pdnsd-1.2.9a-par.drv-0/pdnsd-1.2.9a/src'
  make[1]: *** [Makefile:270: all-recursive] Error 1
  make[1]: Leaving directory '/private/tmp/nix-build-pdnsd-1.2.9a-par.drv-0/pdnsd-1.2.9a'
  make: *** [Makefile:202: all] Error 2
```

@ofborg build pdnsd

--- a/pkgs/tools/networking/pdnsd/default.nix
+++ b/pkgs/tools/networking/pdnsd/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl }:
+{ lib, stdenv, fetchurl, fetchpatch }:
 
 stdenv.mkDerivation rec {
   pname = "pdnsd";
@@ -9,7 +9,16 @@ stdenv.mkDerivation rec {
     sha256 = "0yragv5zk77a1hfkpnsh17vvsw8b14d6mzfng4bb7i58rb83an5v";
   };
 
-  patchPhase = ''
+  patches =
+    # fix build with linux headers >= 5.13
+    lib.optional stdenv.isLinux
+      (fetchpatch {
+        name = "fix-build-linux-headers-gte-5.13.patch";
+        url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/net-dns/pdnsd/files/pdnsd-1.2.9a-linux-5.13_build_fix.patch?id=7ce35657f269c3b7016e8940ad36e59cf06e12a4";
+        hash = "sha256-Sh/0ZyiQpDvFZOWE9OCQ9+ocXurjzJvrE4WNWaGwAwk=";
+      });
+
+  postPatch = ''
     sed -i 's/.*(cachedir).*/:/' Makefile.in
   '';
 

